include/v4l2_camera/rate_bound_status.hpp

```diff
@@ -0,0 +1,281 @@
+// Copyright 2025 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RATE_BOUND_STATUS_HPP_
+#define RATE_BOUND_STATUS_HPP_
+
+#include <diagnostic_updater/diagnostic_updater.hpp>
+#include <diagnostic_msgs/msg/diagnostic_status.hpp>
+
+#include <iomanip>
+#include <sstream>
+#include <variant>
+#include <optional>
+#include <mutex>
+#include <stdexcept>
+#include <chrono>
+
+namespace custom_diagnostic_tasks
+{
+/**
+ * \brief A structure that holds the constructor parameters for the
+ * RateBoundStatus class.
+ */
+struct RateBoundStatusParam
+{
+  RateBoundStatusParam(const double min_freq, const double max_freq)
+      : min_frequency(min_freq), max_frequency(max_freq){}
+
+  double min_frequency;
+  double max_frequency;
+};
+
+/**
+ * \brief Diagnostic task to monitor the interval between events.
+ *
+ * This diagnostic task monitors the difference between consecutive events,
+ * and creates corresponding diagnostics. This task categorize observed intervals into
+ * OK/WARN/ERROR according to the value ranges passed via constructor arguments
+ */
+class RateBoundStatus : public diagnostic_updater::DiagnosticTask
+{
+private:
+  // Helper struct to express state machine nodes
+  struct StateBase
+  {
+    StateBase(const unsigned char lv, const std::string m)
+        : level(lv), num_observation(1), msg(m) {}
+
+    unsigned char level;
+    size_t num_observation;
```

Comment:
I think that `num_observations` (plural) would be more natural

Reply:
These kinds of suggestions are super helpful for me! I fixed this in ccd1e06